CRM/Civicase/Event/Listener/CaseTypeCategoryIsActiveToggler.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+use CRM_Civicase_Factory_CaseTypeCategoryEventHandler as CaseTypeCategoryEventHandlerFactory;
+use Civi\API\Event\RespondEvent;
+
+/**
+ * Class CRM_Civicase_Event_Listener_CaseTypeCategoryIsActiveToggler.
+ *
+ * Contains extra processing for case type category when it is
+ * enabled / disabled, e.g. show / hide related menu item.
+ */
+class CRM_Civicase_Event_Listener_CaseTypeCategoryIsActiveToggler {
+
+  /**
+   * Update case type category related objects on category update.
+   *
+   * @param \Civi\API\Event\RespondEvent $e
+   *   Event data.
+   */
+  public static function onRespond(RespondEvent $e) {
+    $apiRequest = $e->getApiRequest();
+    if ($apiRequest['entity'] != 'OptionValue' || $apiRequest['action'] != 'create') {
+      return;
+    }
+    if (!self::checkReferrer() || !self::checkOptionGroup($apiRequest['params']['option_group_id'])) {
+      return;
+    }
+
+    $handler = CaseTypeCategoryEventHandlerFactory::create();
+    $handler->onUpdate(
+      $apiRequest['params']['id'],
+      isset($apiRequest['params']['is_active']) ? $apiRequest['params']['is_active'] : NULL,
+      isset($apiRequest['params']['icon']) ? $apiRequest['params']['icon'] : NULL
+    );
+  }
+
+  /**
+   * Checks the referer page for this request.
+   *
+   * This check may be used for the sake of performance. As we have to listen
```

Comment:
s/This check may be used for the sake of performance/This check is used for the sake of performance